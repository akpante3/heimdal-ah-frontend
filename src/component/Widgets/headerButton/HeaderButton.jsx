import React from 'react';
import Button from '../../ui/buttons/Button';

const HeaderButton = () => (
  <div className='header-button-group'>
    <Button className="header-button" type='signup' label='Signup' />
    <Button type='login' label='Login' />
  </div>
);

export default HeaderButton;
