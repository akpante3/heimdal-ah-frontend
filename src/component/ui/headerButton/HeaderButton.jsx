import React from 'react';
import ModalButton from '../Modal/Index';
import AuthContainer from '../../../container/AuthContainer';
import SignInForm from '../Forms/Signin/SignInForm';

const HeaderButton = () => (
  <div className='header-button-group'>
    <ModalButton type='signup' label='Signup'>
      Sign up
    </ModalButton>
    <ModalButton type='signup' label='Login'>
      <AuthContainer>
        {data => <SignInForm {...data} />}
      </AuthContainer>
    </ModalButton>
  </div>
);

export default HeaderButton;
