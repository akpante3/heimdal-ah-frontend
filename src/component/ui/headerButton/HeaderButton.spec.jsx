/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import React from 'react';
import toJson from 'enzyme-to-json';
import HeaderButton from './HeaderButton';

describe('testing the Header Button', () => {
  it('should render the image', () => {
    const wrapper = shallow(<HeaderButton />);
    expect(wrapper.length).toBe(1);
  });

  it('should create a snapshot of the logo', () => {
    const tree = shallow(<HeaderButton />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
