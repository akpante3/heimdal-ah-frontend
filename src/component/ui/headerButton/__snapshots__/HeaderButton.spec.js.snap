// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the Header Button should create a snapshot of the logo 1`] = `
<div
  className="header-button-group"
>
  <TriggerModal
    label="Signup"
    type="signup"
  >
    Sign up
  </TriggerModal>
  <TriggerModal
    label="Login"
    type="signup"
  >
    <Connect(AuthContainer)>
      <Component />
    </Connect(AuthContainer)>
  </TriggerModal>
</div>
`;
